vercel/src/lib/AsciiPreview.svelte

```diff
@@ -0,0 +1,83 @@
+<script lang="ts">
+  import Chip from './Chip.svelte';
+
+  // A unique value for identifying this preview
+  export let key: number;
+  export let name: string;
+  export let ansi: string[];
+  export let hex: string[] | null = null;
+  export let chip: string;
+  export let ascii: string = '';
+
+  let dark = true;
+  let trueColor = hex != null;
+
+  $: html = ascii
+    .split('\n')
+    .map((line) => {
+      // TODO Clean up, this is hard to read
+      let spanCount = 0;
+      const htmlLine = line.replace(/\{(\d+)\}/g, (_match, index) => {
+        const i = Number.parseInt(index, 10);
+        const spanText = trueColor
+          ? `<span style="color: ${hex[i]}">`
+          : `<span style="color: ${mapToDefaultTerminalFgColor(
+              ansi[i],
+              dark
+            )}">`;
+        spanCount++;
+        return spanText;
+      });
+      return `${htmlLine}${'</span>'.repeat(spanCount)}`;
+    })
+    .join('\n');
+
+  function mapToDefaultTerminalFgColor(color: string, dark: boolean): string {
+    return color === 'white' && !dark ? 'black' : color;
+  }
+</script>
+
+<div class="title-row">
+  <div class="language-name">
+    <Chip color={chip} width={24} height={24} />
+    <h3>{name}</h3>
+  </div>
+  <div>
+    <input id="dark-checkbox-{key}" type="checkbox" bind:checked={dark} />
```

Comment:
Isn't the `name` unique? maybe we could use it as a key.

Reply:
done in [90fcea0](https://github.com/o2sh/onefetch/pull/701/commits/90fcea0986ef299f2e8cff29a9b03656133a7753)

Reply:
😆 Good point, I had completely forgotten that the name is obviously unique.